servant/src/Servant/API/TypeLevel.hs

```diff
@@ -241,18 +249,53 @@ We might try to factor these our more cleanly, but the type synonyms and type
 families are not evaluated (see https://ghc.haskell.org/trac/ghc/ticket/12048).
 -}
 
+-- ** Fragment
+
+class FragmentUnique api => OnlyOneFragment api
+
+-- | If fragment appeared in API endpoint twice, compile-time error would be raised.
+--
+-- >>> -- type FailAPI = Fragment Bool :> Fragment Int :> Get '[JSON] NoContent
+-- >>> instance OnlyOneFragment FailAPI
+-- ...
+-- ...Only one Fragment allowed per endpoint in api...
+-- ...
+-- ...In the instance declaration for...
+instance OnlyOneFragment (Verb m s ct typ)
```

Comment:
btw, I just realized you need this instance for `UVerb` as well, right?

Reply:
Fixed.